lib/node_modules/@stdlib/blas/base/sspr/src/sspr.c

```diff
@@ -0,0 +1,108 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/base/sspr.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/strided/base/stride2offset.h"
+
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix supplied in packed form.
+*
+* @param order    storage layout
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
+* @param N        number of elements along each dimension of `A`
+* @param alpha    scalar
+* @param x        first input array
+* @param strideX  `x` stride length
+* @param AP       packed form of a symmetric matrix `A`
+*/
+void API_SUFFIX(c_sspr)( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT N, const float alpha, const float *X, const CBLAS_INT strideX, float *AP ) {
+	CBLAS_INT ox;
+
+	if ( N == 0 || alpha == 0.0f ) {
+		return;
+	}
+	ox = STDLIB_BLAS_BASE_STRIDE2OFFSET( N, strideX );
+	API_SUFFIX(c_sspr_ndarray)( order, uplo, N, alpha, X, strideX, ox, AP, 1, 0 );
+	return;
+}
+
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix supplied in packed form using alternative indexing semantics.
+*
+* @param uplo      specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
+* @param N         number of elements along each dimension of `A`
+* @param alpha     scalar
+* @param x         first input array
+* @param strideX   `x` stride length
+* @param offsetX   starting index of `x`
+* @param AP        packed form of a symmetric matrix `A`
+* @param strideAP  stride of the first dimension of `AP`
+* @param offsetAP   starting index of `AP`
+*/
+void API_SUFFIX(c_sspr_ndarray)( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT N, const float alpha, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX, float *AP, const CBLAS_INT strideAP, const CBLAS_INT offsetAP ) {
```

Comment:
Same comments regarding `X` capitalization in docs. If not desired in docs, lowercase here and in the implementation.